### PR TITLE
Mark stats-related record fields as volatile

### LINKED ISSUE #6885: LocalMapStatsTest.testGetAsyncAndHitsGenerated
```
Error Message

expected:<100> but was:<99>
Stacktrace

java.lang.AssertionError: expected:<100> but was:<99>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.map.LocalMapStatsTest$2.run(LocalMapStatsTest.java:100)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:835)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:849)
	at com.hazelcast.map.LocalMapStatsTest.testGetAsyncAndHitsGenerated(LocalMapStatsTest.java:95)
```

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-pr-builder/com.hazelcast$hazelcast/17122/testReport/junit/com.hazelcast.map/LocalMapStatsTest/testGetAsyncAndHitsGenerated/

### PR DESCRIPTION
**Pull request by metanet**
* Stats related fields are updated from partition thread and being read from user thread.

Fixes #6885

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsTest.java [hunk 1]
```diff
@@ -342,7 +345,7 @@ private MemberGroupConfig createGroupConfig(String[] addressArray) {
     }
 
     private <K, V> IMap<K, V> getMap() {
-        HazelcastInstance instance = createHazelcastInstance(getConfig());
+        HazelcastInstance instance = createHazelcastInstance(new Config().setProperty("hazelcast.logging.type", "log4j"));
```
Comment (jerrinot) [inline hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsTest.java]: is this a left-over?

Comment (metanet) [inline hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsTest.java]: yep. fixing it

**Comment by gurbuzali** [comment]
:+1:

**Comment by jerrinot** [comment]
:+1:

**Comment by metanet** [comment]
thanks @jerrinot for pointing out the left-over code. This is beauty of sending PRs :)

File: hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsTest.java (hunk 1 above)
Comment (ahmetmircik) [inline hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsTest.java]: no it is not a leftover. It is used to override configs.

Comment (ahmetmircik) [inline hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsTest.java]: Sorry, wanted to state it should be like this  `HazelcastInstance instance = createHazelcastInstance(getConfig())`

**Comment by ahmetmircik** [comment]
:+1:

**Comment by metanet** [comment]
run-lab-run

**Comment by metanet** [comment]
run-lab-run

**Comment by metanet** [comment]
run-lab-run

**Comment by metanet** [comment]
run-lab-run

**Comment by metanet** [comment]
run-lab-run

**Comment by hasancelik** [comment]
run-lab-run

**Pull request merged**